inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/ActiveLearningServiceImpl.java

```diff
@@ -91,14 +91,15 @@ public boolean isSuggestionVisible(LearningRecord aRecord)
         List<SuggestionGroup> suggestions = getSuggestions(user,
                 aRecord.getLayer());
         for (SuggestionGroup listOfAO : suggestions) {
-            if (listOfAO.stream().anyMatch(suggestion -> 
-                    suggestion.getDocumentName().equals(aRecord.getSourceDocument().getName()) && 
-                    suggestion.getFeature().equals(aRecord.getAnnotationFeature().getName()) && 
-                    suggestion.getLabel().equals(aRecord.getAnnotation()) && 
-                    suggestion.getBegin() == aRecord.getOffsetCharacterBegin() && 
-                    suggestion.getEnd() == aRecord.getOffsetCharacterEnd() &&
-                    suggestion.isVisible())
-            ) {
+            if (listOfAO.stream().anyMatch(suggestion -> suggestion.getDocumentName()
+                    .equals(aRecord.getSourceDocument().getName())
+                    && suggestion.getFeature().equals(aRecord.getAnnotationFeature().getName())
```

Comment:
I would suggest introducing a method `AnnotationSuggestion.labelEquals(otherSuggestion)` which returns true if both labels are null or if both labels are equal instead of repeating the complicated comparison again and again throughout the code.

Reply:
You were right, it's a lot nicer now :)